File: ordenação/JavaScript/shell_sort.js
let vetor = [5, 3, 8, 4, 2];

let tamanho = vetor.length;

// Começa com metade do tamanho
let gap = Math.floor(tamanho / 2);

// Continua enquanto o gap for maior que zero
while (gap > 0) {

    for (let i = gap; i < tamanho; i++) {

        // Guarda o elemento atual
        let temp = vetor[i];
        let j = i;

        // Move elementos maiores
        while (j >= gap && vetor[j - gap] > temp) {
            vetor[j] = vetor[j - gap];
            j -= gap;
        }

        // Coloca o elemento na posição correta
        vetor[j] = temp;
    }

    // Divide o gap pela metade
    gap = Math.floor(gap / 2);
}

console.log("Vetor ordenado:", vetor);

File: ordenação/Python/shell_sort.py
vetor = [5, 3, 8, 4, 2]

tamanho = len(vetor)

# Começa com metade do tamanho
gap = tamanho // 2

# Continua enquanto o gap for maior que zero
while gap > 0:

    for i in range(gap, tamanho):

        # Guarda o elemento atual
        temp = vetor[i]
        j = i

        # Move elementos maiores
        while j >= gap and vetor[j - gap] > temp:
            vetor[j] = vetor[j - gap]
            j -= gap

        # Coloca o elemento na posição correta
        vetor[j] = temp

    # Divide o gap pela metade
    gap //= 2

print("Vetor ordenado:", vetor)
